src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -2281,6 +2281,10 @@ static int __icap_download_bitstream_axlf(struct platform_device *pdev,
 	ICAP_INFO(icap, "incoming xclbin: %pUb\non device xclbin: %pUb",
 		&xclbin->m_header.uuid, &icap->icap_bitstream_uuid);
 
+	err = xocl_cmc_release_access(xdev);
```

Comment:
is this needed when we downloaded xclbin? i thought we only need it for plp download?

Reply:
my understanding is that we need this for blp/plp and ulp, please correct me if I am wrong.

Reply:
With u50, CMC might be accessing user region. In this case, the accessing need to be stopped before downloading xclbin.

Reply:
But, we already have AXI gate freeze and free to protect ULP. I guess that is not protecting CMC from accessing it?

Reply:
It's 'urgly' for now, we would clean this and axi freeze and future ert freeze together in next PR.